Core/include/Acts/EventData/SpacePointProxyIterator.hpp

```diff
@@ -0,0 +1,62 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/SpacePointProxy.hpp"
+#include "Acts/Utilities/Holders.hpp"
+
+namespace Acts {
+
+template <typename container_t>
+class SpacePointProxyIterator {
+ public:
+  using ContainerType = container_t;
+  using ProxyType = typename container_t::SpacePointProxyType;
+
+  using iterator_category = std::random_access_iterator_tag;
+  using value_type = ProxyType;
+  using difference_type = std::ptrdiff_t;
+  using pointer = value_type*;
+  using reference = value_type&;
+
+ public:
+  // Constructors
+  SpacePointProxyIterator(container_t&& container, std::size_t index) = delete;
+  SpacePointProxyIterator(const container_t& container, std::size_t index);
+
+  SpacePointProxyIterator& operator++();
+  SpacePointProxyIterator& operator--();
+  SpacePointProxyIterator operator++(int);
+  SpacePointProxyIterator operator--(int);
+
+  bool operator==(const SpacePointProxyIterator& other) const;
+  bool operator!=(const SpacePointProxyIterator& other) const;
+  bool operator<(const SpacePointProxyIterator& other) const;
+  bool operator>(const SpacePointProxyIterator& other) const;
+  bool operator<=(const SpacePointProxyIterator& other) const;
+  bool operator>=(const SpacePointProxyIterator& other) const;
+
+  SpacePointProxyIterator& operator+=(const std::size_t offset);
+  SpacePointProxyIterator& operator-=(const std::size_t offset);
+
+  SpacePointProxyIterator operator+(const std::size_t offset) const;
+  SpacePointProxyIterator operator-(const std::size_t offset) const;
```

Comment:
do you have a strong preference for removing the `const`? I like to mark as `const` what does not have to be modivied to make it explicit.

kinda obvious from the functions names ... but still

Reply:
No strong preference - this is just something the compiler usually points out to me because `const` on parameters does not do anything on the declaration but only on the definition